### PR TITLE
added several unit tests to iss218

### PR DESCRIPTION
**Pull request by MatakuT**

### CODE REVIEW

**Comment by david-yz-liu** [comment]
@MatakuT Your new tests seem to be failing (see Travis build)

**Comment by MatakuT** [comment]
I think I forgot to upload one important file

**Comment by david-yz-liu** [comment]
@MatakuT Okay, it would be good to have tests which actually check the value of the time returned by the methods.

**Comment by MatakuT** [comment]
The time returned is always different, so I cannot compare it to a fixed string. If that is what you mean by checking the value

**Comment by david-yz-liu** [comment]
Even though the methods probably use the current time, which does change, it would be acceptable to check the computed time to within a certain tolerance (even in hours).

**Comment by david-yz-liu** [comment]
There seems to be a failing test.

**Comment by MatakuT** [comment]
That test had something to do with FactoryGirl naming conventions. They supposed to be unique for all tests. I assume that it is failing because of group names not being unique. Will see if my commit will fix that

File: app/models/submission_rule.rb [hunk 1]
```diff
@@ -27,13 +27,13 @@ def get_collection_time
...
   end
-
+  
   def calculate_collection_time
     assignment.latest_due_date + hours_sum.hours
   end
 
   def calculate_grouping_collection_time(grouping)
-    if grouping.inviter.section
+    if !grouping.inviter.nil?
```
Comment (david-yz-liu) [inline app/models/submission_rule.rb]: @MatakuT This change breaks existing functionality. SectionDueDate should only be queried when a grouping is part of a section.

**Pull request merged**